### ISSUE TITLE
The Future of SquadMaps

### ISSUE THREAD

**Issue by mahtoid** [Napster653/SquadMaps#34]
Hello! My name is Matt (@mahtoid) and I'll be taking over development of SquadMaps alongside Tim (@w4rum) from Squadlanes.

SquadMaps has sadly been inactive from development and maintenance for quite a while. Napster has done a great effort to put SquadMaps on the map, and to build it into what it is today. However, he has sadly been taken up by real life and can no longer put in the time commitment into keeping SquadMaps updated. 

With that said, both myself and Tim have already had a great discussion on what to do with SquadMaps to bring it back to life. Although a little patience is needed, especially when working with Tim.

### Road Map

 - Within the next week (or perhaps after the 4.2 drop): 
    - There will be an update on SquadMaps with the latest maps, vehicles, layers etc.
    
 - From early March:
    - Development will begin on SquadMaps 2.0
      _It's extremely too early to say when we plan to release this, however we look to update the entire site. Modernising the design, switching from HTML to React - and finding an optimised way (using Tim's Squadlanes magic) to generate and display the latest maps and map information._ 
    
 - Into the future and beyond:
    - We will look at keeping SquadMaps updated to the best of our abilities
    - We will work on your suggestions into shaping SquadMaps to be a great place
    
---

We are very open to hearing your suggestions on what you would like on SquadMaps, whether it's a search component (don't worry, we're already on this!) or additional Squad content to be displayed. Feel free to drop a comment below, or even give suggestions in-game.

💜

**Comment by Ivoz** [comment]
But you killed it... 😭pls update CNAME's / DNS pointers!

**Comment by mahtoid** [comment]
@Ivoz killed it already 😭 

Napster transferred the domain across to me last night, and I was hoping the DNS would take. However I'll be resolving this during the day, I do apologize.

**Comment by TomNedry** [comment]
@mahtoid: A huge thanks to for stepping in... very appreciated!

**Comment by mahtoid** [comment]
@TomNedry glad to hear it. SquadMaps is definitely in need for some loving! :smile:

**Comment by mahtoid** [comment]
_oof. It's been a minute._

### A bit of an update...

Good news and bad news! Since the original message back in January, a lot has gone on - but mostly in real life. When it came to February, I had started on the rough prototype for the new version of Squad Maps. I got everything ready for when Tim would become inevitably free in March, as planned... or so I thought.

March came and went, April came and went, May came and went, June even came and went, July came - and with July, so did the news. Tim had ~~died~~ got a job. Tim actually got a job a few months before, and was battling with availability - but in July, he pulled the plug and stepped away from the project. 

This left things a little on the swings, especially seeing as the SquadLanes extractor was going to provide a fatal part of the process. However, Squad Pipeline (which we use in the current version) still exists and provides enough detail that we could get something going. 

That left another issue, I also have very little time - so as you may have saw on the Discord, or the website - I placed banners asking for help. I had quite a few messages, a lot from people wishing to help but had very little or no knowledge of React - just fans of the project and wishing for its continuation. 

However, on the 24th August, I received a message from @1KINGO1 - and since then SquadMaps v2 became a rocket ship! I also reached out to @Shanomac99 (Squad Pipeline developer) - who had mentioned to me about an upcoming Squad Pipeline rewrite - however mentioned that once the update was complete, he would happily assist with SquadMaps v2. 

So with that all said, we are moving forward once again, and at a very decent pace. When do we expect to release? There is no confirmed date yet, but it will definitely be in the next couple of months.

---

_tl;dr - Tim died, and also SquadMaps v2 soon:tm:_

**Comment by mahtoid** [comment]
### SquadMaps v2 Beta!

Only good news this time round, I promise. Some people already cracked the link, but I am now sharing that SquadMaps v2 beta is open with the latest Squad information (from version 6). 

SquadMaps v2: [https://beta.squadmaps.com](https://beta.squadmaps.com)

Take a look, and enjoy! If you happen to stumble in to any bugs or have some feedback you would like to give - then let me know either here or [in the Discord](https://discord.mahto.id).

_Please keep in mind that some functions within the site are still actively being worked on, so if the site goes dark for 5 minutes - it just means I am doing a deploy, and something new is about to appear._

**Comment by D4n2021** [comment]
Amazing!!!!! But can you please also add links to the faction articles on the official Squad Fandom, so that we can click on the flag icons of any layer and make that faction page open in a new browser-tab?

**Comment by mahtoid** [comment]
@D4n2021 out of interest, what Faction information is it that you find useful? I saw something on Reddit for someone asking for a list of Faction vehicles - however, I am keen to know your thoughts. Also, what other information may Squad Fandom be missing, which you are interested in having?

**Comment by D4n2021** [comment]
Quite generally **all** vehicles and kits (handheld weapons) and static weapons (TOW etc.) of the factions. Apart from the information that Fandom already has, I am always curious regarding acceleration and speed statistics of Squad vehicles, so I created my own statistics wiki a few months ago 😉 Feel free to check it out! https://squad-vehicles.fandom.com/wiki/Squad_vehicles_Wiki

**Comment by D4n2021** [comment]
(Here is the backup page I sadly had to create, due to unbelievable auto-deletion of my wiki due to allegedly being "too inactive".... Happened a few weeks ago https://sites.google.com/view/squad-vehicles )

**Comment by mahtoid** [comment]
How did you gather this data, did you manually fetch it from the SDK or do you have a scrapper?

**Comment by D4n2021** [comment]
Didn't you read the introduction at the top of the wiki? 😆 I explained it all in there. I gathered it **manually**, on the runway of Tallil airfield! 😂

**Comment by mahtoid** [comment]
Hey!

Can you believe it's been exactly a year (and 2 days) since I made this 'The Future of SquadMaps'? Well, the day is finally here! SquadMaps v2 is out with a fresh new look, featuring RAAS and Lane Prediction, detailed Map and Layer info, and a lot more – and guess what? There's even more awesomeness on the horizon.

A massive shoutout to **_KINGO_**, who stepped up big time with development while I was swamped, and props to **_grey275_** for his artistic touch in the redesign and bringing in the lane prediction from SquadLanes.

But it's not just us – a heartfelt thank you to all of you! _Melle_, _ButterBread_, _Matador_, _Martinez_, _Dral_, _BlueMoonTR_, and many others have been incredible in helping spot bugs and suggesting cool features along the way.

You can explore it here: [https://squadmaps.com/](https://squadmaps.com/)

**Now that SquadMaps v2 is out, what's next, you ask?**

We’ve got quite a few things planned, from a particular type of calculator, vehicle spawn locations (suggested by _FactualComment_), and a few other big things.

However, before diving into these tasks, we've decided to take a breather. We want to check out the forthcoming ‘Map Voting’ update to see the work OWI has done, understand its potential impact, and figure out how we can adapt to it.

So, here's the deal – we will assess the situation after the 'Map Voting' update, see where OWI (Offworld Industries) takes things, and then chart our next move. Your support has been incredible, and we want to make sure whatever comes next is just as awesome.

As always, if you come across any bugs or have suggestions for new features, feel free to leave a comment below or hop into the Discord! Your feedback is invaluable to us. 💜

**Comment by angellfall** [comment]
You should add a mortar calculator in top of the lanes with a FOB planner. This would include all essentials under one site.

**Comment by mahtoid** [comment]
@angellfall hey, nice suggestions! We do have some stuff in the works, but as mentioned on the post above - we just want to see how OWI implements their new 'Unit' Map Voting stuff first before we continue on the "fun" stuff.

Out of interest, but 'FOB planner', do you mean our own variation of wottactics?

**Comment by mahtoid** [comment]
## SquadMaps Update (Squad v7.1)

Sorry on the delay, but we are finally here with the latest Squad version (v7.1) - along with some much needed post-release quality of life fixes. Details are below...
### Solved
- Jensens Range displaying the wrong team flags
- Map re-centering too early when sidebars open/close
- ZH translations
- Vehicle display inappropriate de-duping
- Unable to view all vehicles on the vehicle list in smaller resolution
### Improved
- Layer selection: RAAS, Invasion, everything else...
- Hotkeys: DE language users can now use `/` and `-` for search shortcut
- AAS Team Names: 'Team Main' changed to display the appropriate 1 or 2
- QoL: Logo and 'Reset' button will now reset filters and search bar

**Comment by D4n2021** [comment]
Hi, can we donate anywhere to support the hosting cost of squadmaps.com?

**Comment by D4n2021** [comment]
Ah nevermind, just realized there is a sponsor option at the bottom, nice!

**Comment by mahtoid** [comment]
@D4n2021 Hey, sorry for the delay - thank you for the support! Much appreciated 💜
